src/core/utils.js

```diff
@@ -77,7 +77,7 @@ function hexToArray(h) {
 
 function arrayToHex(a) {
   return a.map(function(byteValue) {
-    const hex = byteValue.toString(16);
+    const hex = byteValue.toString(16).toUpperCase();
```

Comment:
We use upper case everywhere right, so may as well just make this upper case off the bat. It's better to do this per byte, or once joined?